src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -90,11 +90,14 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
             }
 
             // Validate the document
-            var errors = document.Validate(_settings.RuleSet);
-            foreach (var item in errors)
+            if (_settings.RuleSet != null && _settings.RuleSet.Rules.Count > 0)
```

Comment:
>&& _settings.RuleSet.Rules.Count > 0 [](start = 42, length = 36)

just checking. Is this constraint needed?

Reply:
It's there for performance.  Otherwise we walk the entire graph and do no validation.

---
In reply to: [182166250](https://github.com/Microsoft/OpenAPI.NET/pull/241#discussion_r182166250) [](ancestors = 182166250)